src/utils/index.js

```diff
@@ -0,0 +1,56 @@
+'use strict';
+
+import crypto from 'crypto';
+
+/**
+ * Rejects the request.
+ * @returns {Promise} error - Returns a Promise with the details for the wrong request.
+ */
+export function rejectValidation(module, param) {
+  return Promise.reject({
+    status: 0,
+    message: `The ${module} ${param} is not valid or it was not specified properly`
+  });
+}
+
+export const bodyTypes = {
+  BUFFER: 'BUFFER',
+
+  /**
+   * @param {Object} body - The file body whose type we need to determine
+   * @return {string} One of bodyTypes.BUFFER, or null
+   */
+  get(body) {
+    if (Buffer.isBuffer(body)) {
+      return bodyTypes.BUFFER;
+    }
+    return null;
+  }
+};
+
+/**
+ * Returns file size
+ * @param {Object} file File
+ * @returns {Number} The amount of data that can be read from the file
+ */
+export function getLength(file) {
+  const { body, length } = file;
+  const bodyType = bodyTypes.get(body);
+
+  if (bodyType === bodyTypes.BUFFER) {
+    return body.length;
```

Comment:
What would `length` be in this case? And don't we always need the body length anyway for our purpose?

Reply:
Is the file size. Or, in Node terms, the length of the buffer.

Reply:
Yes, but I mean, what's the difference with the length from `const { body, length } = file` which will be returned if this if condition is not true?

Reply:
This is more of a left over of the old implementation. I'm changing things around to support stream objects.

Reply:
Going back to this, the `body.length` if just a fallback. If the `body` doesn't have that property, the `uploadFile` method will raise an exception. We cannot upload files that have `0` size.

Think about it as a fallback.

Reply:
Wouldn't we want to (re)throw that exception instead of returning `length` (which would be 0) in that case?